snake-odin/src/snake.odin

```diff
@@ -0,0 +1,109 @@
+package main
+
+import "w4"
+import "core:runtime"
+
+Point :: struct {
+	X: i32,
+	Y: i32,
+}
+
+are_points_colliding :: proc "c" (a, b : Point) -> bool {
+	return a.X == b.X && a.Y == b.Y 
+}
```

Comment:
Fun fact: this procedure is not needed since `Point` is a comparable struct.

Another fun fact:
```odin
Point :: distinct [2]i32

p: Point
p.x = 123 // this works
```

Reply:
Cool. I didn't know that. I'll use this instead.